### ISSUE TITLE
lastTransitionTime format for "Ready" condition of Kafka objects cannot be deserialized

### ISSUE THREAD

**Issue by antoineco** [strimzi/strimzi-kafka-operator#4538]
**Describe the bug**

The timestamp of the Ready condition's `lastTransitionTime` field in `Kafka` objects is serialized in a format that Kubernetes reflectors can not handle:

```yaml
  conditions:
  - lastTransitionTime: 2021-03-09T10:32:01+0000
    status: "True"
    type: Ready
```

Logs from a controller (specifically from the Knative project, but could apply to anything):

`E0309 11:45:11.706744   16754 reflector.go:127] knative.dev/pkg/apis/duck/typed.go:67: Failed to watch *v1.Source: failed to list *v1.Source: parsing time "2021-03-09T10:32:01+0000" as "2006-01-02T15:04:05Z07:00": cannot parse "+0000" as "Z07:00"`

In the example above, the timestamp should have been `2021-03-09T10:32:01Z`.

**To Reproduce**

1. Create a `Kafka` and let the operator update its status conditions.
2. Copy the value of the generated `lastTransitionTime` timestamp of the Ready status condition.
3. Confirm Kubernetes can't unmarshal it: https://play.golang.org/p/1mMMHBuRobA (if the playground times out, run locally)

**Expected behavior**

The operator uses the same RFC 3339 format for all timestamp fields, and this format can be [deserialized by Kubernetes](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Time.UnmarshalJSON).

Example of another status condition set by the Strimzi operator, this time with a valid RFC 3339 format:

```yaml
  - lastTransitionTime: "2021-03-09T10:30:11.673912Z"
    message: In API version kafka.strimzi.io/v1beta1 the object kafkaListeners at
      path spec.kafka.listeners.kafkaListeners has been deprecated. This object has
      been replaced with GenericKafkaListener.
    reason: DeprecatedObjects
    status: "True"
    type: Warning
```

**Environment (please complete the following information):**
 - Strimzi version: 0.21.1
 - Installation method: `kubectl apply -f strimzi-cluster-operator-0.21.1.yaml`
 - Kubernetes cluster: v1.20.4
 - Infrastructure: GKE

**YAML files and logs**

_not relevant_

**Additional context**

_none_

**Comment by scholzj** [comment]
In which resource did you saw the wrong timestamp?

**Comment by scholzj** [comment]
Ok, looking at the code, I guess the `Kafka` one.

**Comment by scholzj** [comment]
Ok, I opened a PR to have it fixed in 0.22. Thanks for raising this.
